marketplace/sensitive-language/src/language-checker.js

```diff
@@ -0,0 +1,130 @@
+'use strict';
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Icon,
+  Typography,
+  Subheading,
+  SectionHeading,
+  Paragraph
+} from '@contentful/forma-36-react-components';
+import { documentToPlainTextString } from '@contentful/rich-text-plain-text-renderer';
+import alex from 'alex';
+
+import { NoIssues } from './no-issues.js';
+import { Message } from './message.js';
+
+function checkContent(isRichText, value, config = {}) {
+  const { messages } = isRichText
+    ? alex.text(documentToPlainTextString(value), config)
+    : alex.markdown(value, config);
+
+  return messages;
+}
+
+export class LanguageChecker extends React.Component {
+  constructor() {
+    super();
+    this.state = { messageMap: {} };
+  }
+
+  componentDidMount() {
+    const { entry, fieldsToCheck, alexConfig } = this.props;
+
+    fieldsToCheck.forEach(fieldDefinition => {
+      const fieldId = fieldDefinition.id;
+      const field = entry.fields[fieldId];
+      const currentValue = field.getValue();
+      const isRichText = fieldDefinition.type === 'RichText';
+
+      if (currentValue) {
+        const messages = checkContent(isRichText, currentValue, alexConfig);
+
+        this.setState(state => {
+          const messageMap = state.messageMap;
+
+          messageMap[fieldId] = messages;
+
+          return { messageMap };
+        });
+      }
+
+      field.onValueChanged(value => {
+        const messages = checkContent(isRichText, value, alexConfig);
+
+        this.setState(state => {
+          const messageMap = state.messageMap;
+
+          messageMap[fieldId] = messages;
+
+          return { messageMap };
```

Comment:
I wouldn't mutate state objects in place.

```js
return { messageMap: { ...messageMap, [fieldId]: messages } }
```

Also exactly the same state setting is repeated twice - could be a method (`this.updateMapForField(fieldId, messages)`